From 151aa8bbdedfc0a95bea9b30b87e14776282a2c2 Mon Sep 17 00:00:00 2001
From: Devarsh Thakkar <devarsht@xilinx.com>
Date: Tue, 20 Sep 2016 17:47:58 +0530
Subject: [PATCH 02/24] configure : Create base for zynqultrascaleplus devices

Generate separate gstomx.conf for zynqultrascaleplus device when
with-omx-target set  zynqmpsoc+ as shown below :

Usage :
./configure --prefix=<path_to_prefix> --with-omx-target=zynqultrascaleplus

Signed-off-by: Devarsh Thakkar <devarsht@xilinx.com>
Upstream Status: Pending
---
 config/zynqultrascaleplus/Makefile.am | 6 ++++++
 config/zynqultrascaleplus/gstomx.conf | 0
 configure.ac                          | 9 +++++++--
 3 files changed, 13 insertions(+), 2 deletions(-)
 create mode 100644 config/zynqultrascaleplus/Makefile.am
 create mode 100644 config/zynqultrascaleplus/gstomx.conf

diff --git a/config/zynqultrascaleplus/Makefile.am b/config/zynqultrascaleplus/Makefile.am
new file mode 100644
index 0000000..01671c2
--- /dev/null
+++ b/config/zynqultrascaleplus/Makefile.am
@@ -0,0 +1,6 @@
+EXTRA_DIST = gstomx.conf
+
+if USE_OMX_TARGET_ZYNQ_USCALE_PLUS
+configdir = $(sysconfdir)/xdg
+config_DATA = gstomx.conf
+endif
diff --git a/config/zynqultrascaleplus/gstomx.conf b/config/zynqultrascaleplus/gstomx.conf
new file mode 100644
index 0000000..e69de29
diff --git a/configure.ac b/configure.ac
index e87d240..7bce920 100644
--- a/configure.ac
+++ b/configure.ac
@@ -221,7 +221,7 @@ AC_ARG_ENABLE(Bsymbolic,
                LDFLAGS="${SAVED_LDFLAGS}"])
 
 AC_ARG_WITH([omx-target],
-        AS_HELP_STRING([--with-omx-target],[Use this OpenMAX IL target (generic, bellagio, rpi)]),
+        AS_HELP_STRING([--with-omx-target],[Use this OpenMAX IL target (generic, bellagio, rpi, zynqultrascaleplus)]),
         [ac_cv_omx_target="$withval"], [ac_cv_omx_target="none"])
 
 ac_cv_omx_target_struct_packing="none"
@@ -237,13 +237,17 @@ case "${ac_cv_omx_target}" in
   bellagio)
     AC_DEFINE(USE_OMX_TARGET_BELLAGIO, 1, [Use Bellagio OpenMAX IL target])
     ;;
+  zynqultrascaleplus)
+    AC_DEFINE(USE_OMX_TARGET_ZYNQ_USCALE_PLUS, 1, [Use Bellagio OpenMAX IL target])
+    ;;
   none|*)
-    AC_ERROR([invalid OpenMAX IL target, you must specify one of --with-omx-target={generic,rpi,bellagio}])
+    AC_ERROR([invalid OpenMAX IL target, you must specify one of --with-omx-target={generic,rpi,bellagio,zynqultrascaleplus}])
     ;;
 esac
 AM_CONDITIONAL(USE_OMX_TARGET_GENERIC, test "x$ac_cv_omx_target" = "xgeneric")
 AM_CONDITIONAL(USE_OMX_TARGET_BELLAGIO, test "x$ac_cv_omx_target" = "xbellagio")
 AM_CONDITIONAL(USE_OMX_TARGET_RPI, test "x$ac_cv_omx_target" = "xrpi")
+AM_CONDITIONAL(USE_OMX_TARGET_ZYNQ_USCALE_PLUS, test "x$ac_cv_omx_target" = "xzynqultrascaleplus")
 
 AC_ARG_WITH([omx-struct-packing],
         AS_HELP_STRING([--with-omx-struct-packing],[Force OpenMAX struct packing, (default is none)]),
@@ -387,6 +391,7 @@ tools/Makefile
 config/Makefile
 config/bellagio/Makefile
 config/rpi/Makefile
+config/zynqultrascaleplus/Makefile
 examples/Makefile
 examples/egl/Makefile
 )
-- 
2.7.4

